### PR TITLE
[SD-3452] - Image cropping improved

### PR DESCRIPTION
**Pull request by Aca-jov**

### CODE REVIEW

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 1]
```diff
@@ -960,6 +960,10 @@
                         $scope.origItem = res;
                         $scope.dirty = false;
                         $scope.item = _.create($scope.origItem);
+                        if (res.cropData) {
+                            $scope.item.hasCropes = true;
```
Comment (sivakuna-aap) [inline client/app/scripts/superdesk-authoring/authoring.js]: spell mistake - should be `hasCrops` without `e`

File: client/app/scripts/superdesk-authoring/authoring.js [hunk 2]
```diff
@@ -1978,7 +1982,13 @@
                 });
 
                 metadata.initialize().then(function() {
+                    scope.item.hasCropes = false;
                     scope.metadata = metadata.values;
+                    _.each(scope.metadata.crop_sizes, function(crop) {
+                        if (scope.item.renditions[crop.name]){
+                            scope.item.hasCropes = true;
+                        }
+                    })
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/authoring.js]: You can just use the built-in [some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method here, more concise:
```js
scope.metadata = metadata.values;
scope.item.hasCrops = scope.metadata.crop_sizes.some(function (crop) {
    return scope.item.renditions[crop.name];
});
```

File: client/app/scripts/superdesk-authoring/views/article-edit.html [hunk 3]
```diff
@@ -97,14 +97,17 @@
...
   <div ng-if="item.type == 'picture'" class="full-preview" sd-ratio-calc>
-      <div sd-item-crops data-item="item"></div>
       <div>
           <span translate>Original</span>
           <div sd-item-rendition data-item="item" data-rendition="baseImage"></div>
       </div>
+      <div sd-item-crops data-item="item"></div>
+      <button id="btnCrop" class="btn btn-medium pull-right" ng-click="applyCrop()" ng-if="metadata.crop_sizes">
+        <span ng-if="!item.hasCropes">APPLY CROPS</span>
+        <span ng-if="item.hasCropes">EDIT CROPS</span>
```
Comment (plamut) [inline client/app/scripts/superdesk-authoring/views/article-edit.html]: `translate`missing here for the two `<span>` elements.

**Comment by plamut** [comment]
Apart from the comments above, a few tests should be added, e.g. if `scope.item.hasCropes` is initialized correctly depending on the `crop_sizes` metadata.

**Comment by mugurrus** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**